### PR TITLE
Ports TG Cloning

### PR DESCRIPTION
**Pull request by SynnGraffkin**
## About The Pull Request

Ports the rest of the TG cloning file, with all the economy, insurance, trauma code removed.

## Why It's Good For The Game

Yeah, there isn't prescanning, but prescans / autoclone invalidates medical when someone dying is just immediately cloned without anything done to retrieve them because they made a scan 5 minutes into the round. Issues in regards to aghosts being cloned should also be resolved, per testing. 

The good news though: 
- Genetics access instead of Head access to modify records. 
- Geneticists can use diskettes to copy or apply UI, UE, and SE to cloning records 
- The cloner will still automatically process a clone when a corpse / brain is brought in
- General code cleanup the lovely people at TG did (such as removing .src) 
- 🦀 Empties still work 🦀

image

## Changelog
:cl:
add: Geneticists can use Diskettes to copy or apply UI, UE, and SE to and from cloning records. 
balance: Geneticists access is the requirement, instead of Head access, to modify cloning records.
balance: Prescanning / Automatic cloning on death removed, due to TG code.
balance: Minimum level for autocloning lowered to tier 3.
code: General code cleanup from TG's end.
/:cl:

### CODE REVIEW

**Comment by SynnGraffkin** [comment]
Not sure why it has conflicts when it is what I changed.

File: code/__HELPERS/records.dm [hunk 1]
```diff
@@ -0,0 +1,3 @@
+/proc/overwrite_field_if_available(datum/data/record/base, datum/data/record/other, var/field_name)
```
Comment (LetterN) [inline code/__HELPERS/records.dm]:
```suggestion
/proc/overwrite_field_if_available(datum/data/record/base, datum/data/record/other, field_name)
```

Comment (LetterN) [inline code/__HELPERS/records.dm]:
```suggestion
/proc/overwrite_field_if_available(datum/data/record/base, datum/data/record/other, field_name)
```

Comment (SynnGraffkin) [reply code/__HELPERS/records.dm]: Why remove the var/ from the field name?

**Comment by AtlasRules** [comment]
>"ports TG cloning"

but

cloning is dead and cobby killed it

**Comment by Dahlular** [comment]
Not really the direction we wanna go, sorry.

**Pull request closed without merging**